pkg/eventlog/util.go

```diff
@@ -0,0 +1,32 @@
+package eventlog
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+)
+
+func SortAttributes(attrs Attributes, filter map[string]bool) (*Attributes, error) {
+	if len(filter) == 0 {
+		return nil, errors.New("filter must be provided")
+	}
+
+	filtered := make(Attributes, 0, len(attrs))
+	for _, attr := range attrs {
+		if _, ok := filter[attr.Key]; ok {
+			filtered = append(filtered, attr)
+		}
+	}
+
+	idx := 0
+	end := len(filter)
+	for end <= len(filtered) {
```

Comment:
Nit: could you add a simple test case for this function? especially this statement needs some context behind.